---
layout: page
title: Cancer Research Resources
author: Alex Gibbs
---

This page is intended to provide cancer researchers with a variety of resources to aid in their research. 
This page will contain the following: Cancer sample databases, Cancer analysis tools, and Cancer Atlases.

***

# Cancer Sample Databases



Here you will find various databases which users can search and download publicly available cancer data.
If this is something you would like to do, i.e. download, process, and analyse raw sequencing data from a specific study, then please contact me! (**GibbsA@cardiff.ac.uk**)

***

### [*Gene Expression Omnibus (GEO)*](https://www.ncbi.nlm.nih.gov/geo/) 

![Gene Expression Omnibus!](/assets/img/geo.png)

Public repository that archives and distributes high-throughput gene expression and other functional genomics data sets. 
It is maintained by the National Center for Biotechnology Information (NCBI) at the National Library of Medicine (NLM).
Link to the [Overview](https://www.ncbi.nlm.nih.gov/geo/info/overview.html). Link to the [FAQ](https://www.ncbi.nlm.nih.gov/geo/info/faq.html). Link on [how to download data](https://www.ncbi.nlm.nih.gov/geo/info/download.html).

***

### [*Array Express*](https://www.ebi.ac.uk/biostudies/arrayexpress)

![Array Express!](/assets/img/arrayexpress.png)

The functional genomics data collection stores data from high-throughput functional genomics experiments, and provides data for reuse to the research community.
Link to the [Help section](https://www.ebi.ac.uk/biostudies/arrayexpress/help).

***

### [*European Nucleotide Archive (ENA)*](https://www.ebi.ac.uk/ena/browser/home)

![European Nucleotide Archive (ENA)!](/assets/img/ena.png)

The European Nucleotide Archive provides a comprehensive record of the worldâ€™s nucleotide sequencing information, covering raw sequencing data, sequence assembly information and functional annotation.
Link to the [support guides page](https://www.ebi.ac.uk/ena/browser/guides).

***
  
### [*Sequence Read Archive (SRA)*](https://www.ncbi.nlm.nih.gov/sra)

![Sequence Read Archive (SRA)!](/assets/img/sra.png)

Sequence Read Archive data, available through multiple cloud providers and NCBI servers, is the largest publicly available repository of high throughput sequencing data.
SRA stores raw sequencing data and alignment information to enhance reproducibility and facilitate new discoveries through data analysis. 
[how to search and download](https://www.ncbi.nlm.nih.gov/sra/docs/sradownload/) link.

***

### [*Zenodo*](https://zenodo.org/)

![Zenodo!](/assets/img/zenodo.png)

Open research repository. Search for cancer and you get list of uploads associated with papers. Every file that the paper used for the publication is uploaded, such as loom files for RNA velocity analysis etc etc.
Link to the [Guides](https://help.zenodo.org/docs/).

***

### [*OmicsDI*](https://www.omicsdi.org/)

![OmicsDI!](/assets/img/omicsdi.png)

Database of sequencing datasets. Home page has lots of quick stats and search bar. Allows user to find databases related to their search and access that data.
Link to [Help section](http://blog.omicsdi.org/).

***

### [*Genomics Data Commons (GDC) Data Portal*](https://portal.gdc.cancer.gov/repository)

![Genomics Data Commons (GDC) Data Portal!](/assets/img/gdc.png)

Search for your favourite gene or cancer to get relevant statistics. There are a number of great tools on this site! Data is also downloadable for further analysis.
Link to the [Documentation](https://docs.gdc.cancer.gov/).

***

### [*Google Dataset Search Tool*](https://datasetsearch.research.google.com/)

![Google Dataset Search Tool!](/assets/img/google.png)

Search for public datasets. 

Link to the [User Support Page](https://datasetsearch.research.google.com/help)

***

### [*PanglaoDB*](https://panglaodb.se/search.html) 

![PanglaoDB!](/assets/img/panglaodb.png)

A database for scRNAseq data downloads, but also you can search a gene across datasets.

Link to [FAQ/support page](https://panglaodb.se/faq.html)

***

### [*Curated Atlas Query R*](https://stemangiola.github.io/CuratedAtlasQueryR/)

![Curated Atlas Query R!](/assets/img/curated-atlas-query-r.png)

A query interface that allow the programmatic exploration and retrieval of the harmonised, curated and reannotated CELLxGENE single-cell human cell atlas.
**Must be run within R.**

***

### [*3CA - Curated Cancer Cell Atlas*](https://www.weizmann.ac.il/sites/3CA/)

![3CA - Curated Cancer Cell Atlas!](/assets/img/3ca.png)

Collected, annotated and analyzed cancer scRNA-seq datasets. [Paper](https://pubmed.ncbi.nlm.nih.gov/37258682/).

***

### [*IMMUcan Database*](https://immucanscdb.vital-it.ch/)

![IMMUcan Database!](/assets/img/immucan.png)

A fully integrated scRNA-seq database exclusively dedicated to human cancer and accessible to nonspecialists. IMMUcan scDB encompasses 144 datasets on 56 different cancer types, annotated in 50 fields containing precise clinical, technological, and biological information.

***

### [*JingleBells*](https://jinglebells.bgu.ac.il/aboutus/)

![JingleBells!](/assets/img/jingle-bells.png)

A repository of standardized single cell RNA-Seq datasets for analysis and visualization at the single cell level.

***

### [*scPortalen Database*](https://single-cell.riken.jp/SCPortalen_Database/Home/)

![scPortalen Database!](/assets/img/scportalen.png)

This database features integration of single cell metadata, cell images and sequence information. Rhe platform is divided into two parts, the single cell dataset generated by the research team, and the single cell dataset published in peer-reviewed papers.

***

### [*Cancer Single-cell Expression Map*](https://ngdc.cncb.ac.cn/cancerscem/)

![Cancer Single-cell Expression Map!](/assets/img/cancer-single-cell-expression-map.png)

Public database dedicating to collecting, analysing, and visualising single-cell RNAseq data of human cancers.

[Database Usage](https://ngdc.cncb.ac.cn/cancerscem/documents#docu_us) link.

***

***

# Cancer Analysis Tools

Here you will find links to various tools that allow the user to explore and analyse cancer data. These tools are mostly online, meaning users only need an internet browser and a stable internet connection. 
However, some are available as downloadable apps or can be run through R or Python. If this something you would like to do but are not sure how, please contact me! (**GibbsA@cardiff.ac.uk**)

*** 

***

### [*Xena*](https://xenabrowser.net/)

![Xena!](/assets/img/xena.png)

This tool allows users to explore functional genomic data sets for correlations between genomnic and/or phenotypic variables. Users can choose a study to explore up to two variables. There is also a *Transcripts* tab which allows the user to visualise the various transcripts of a gene.
Link to the [Help page](https://ucsc-xena.gitbook.io/project/).

***

### [*NCRAS - National Cancer Registration and Analysis Service*](http://www.ncin.org.uk/cancer_information_tools/)

![NCRAS!](/assets/img/ncras.png)

The National Cancer Registration and Analysis Service (NCRAS) collects, quality assures and analyses data on all people living in England who are diagnosed with cancer. It is part of the National Disease Registration Service (NDRS) in NHS Digital (NHSD). There are a plethora of tools on here to exploit this data.


***

### [*Cancer Research UK - Cancer Statistics for the UK*](https://www.cancerresearchuk.org/health-professional/cancer-statistics-for-the-uk)

![CRUK!](/assets/img/cruk.png)

CRUK have a nice, more basic tool on this website to get quick and simple cancer stats.

***

### [*Cancer Research UK - Early Diagnosis Hub*](https://crukcancerintelligence.shinyapps.io/EarlyDiagnosis/)

![CRUK - early diagnosis hub!](/assets/img/cruk-early-diagnosis-hub.png)

CRUK also have an early diagnosis hub tool that allows exploration of the latest cancer early diagnosis data across the UK.

***

### [*OncoLnc*](http://www.oncolnc.org/)

Link TCGA survival data to mRNA, miRNA, or lncRNA expression levels.
Link to [tutorial video](https://www.youtube.com/watch?v=5h5NU15pKoI).

***

### [*CIViC - clinical interpretation of variants in cancer database*](https://civicdb.org/welcome)

![CIViC - clinical interpretation of variants in cancer database!](/assets/img/civic.png)

Primarily a tool for doctors to get the best treatment options for patients with certain mutations. You can search for specific muations/gene variants and the database gives you a load of information on it. Can also look for specific cancers, specific therapies. Really neat database.
Link on [how to use CIVIC](https://civic.readthedocs.io/en/latest/using.html).

***

### [*CrossHub*](https://sourceforge.net/projects/crosshub/) 

Multi-way analysis of RNAseq, miRNAseq and methylome data from the TCGA project. **Python tool**. Generates excel summaries.

***

### [*Cellenics*](https://www.biomage.net/)

![Cellenics!](/assets/img/cellenics.png)

Open source scRNAseq analysis software online tool to analyse public scRNAseq datasets. Its a cloud based tool and everything is done online. Need to sign up for free. Very neat tool. Click on the link and then the Cellenics tab.
Link to the [User Guide](https://www.biomage.net/user-guide).

***

### [*Single Cell Portal*](https://singlecell.broadinstitute.org/single_cell)

![Single Cell Portal!](/assets/img/single-cell-portal.png)

Allows data exploration of curated datasets. scRNAseq and spatial datasets available!
Link to [Help Page](https://singlecell.zendesk.com/hc/en-us).

***

### [*UCSC Cell Browser*](https://cells.ucsc.edu/)

![UCSC Cell Browser!](/assets/img/ucsc-cell-browser.png)

Interactive viewer for scRNAseq expression.
How to use the [website](https://cellbrowser.readthedocs.io/en/master/interface.html).

***

### [*OmicsDI*](https://www.omicsdi.org/)

![OmicsDI!](/assets/img/omicsdi.png)

Database of sequencing datasets. Home page has lots of quick stats and search bar. Allows user to find databases related to their search and access that data.
[Tutorial page](http://blog.omicsdi.org/categories/tutorials/).

***

### [*Dependency Map (DepMap) portal*](https://depmap.org/portal/)

![Dependency Map (DepMap) portal!](/assets/img/depmap-portal.png)

Open access to key cancer dependencies analytical and visualisation tools.
[Documentation page](https://depmap.org/portal/documentation/).

***

### [*ISOexpresso*](http://wiki.tgilab.org/ISOexpresso/)

![ISOexpresso!](/assets/img/isoexpresso.png)

Isoform expression resource for isoform expression analysis in cancer.

***

### [*TCGA Spliceseq*](https://bioinformatics.mdanderson.org/TCGASpliceSeq/index.jsp)

![TCGA Spliceseq!](/assets/img/tcga-spliceseq.png)

Cancer splicing visualised.
[FAQ page](https://bioinformatics.mdanderson.org/TCGASpliceSeq/faq.jsp).

***

### [*Genomics Data Commons (GDC) Data Portal*](https://portal.gdc.cancer.gov/)

![Genomics Data Commons (GDC) Data Portal!](/assets/img/gdc.png)

Search for your favourite gene or cancer to get relevant statistics. There are a number of great tools on this site! Data is also downloadable for further analysis.
Link to the [Documentation](https://docs.gdc.cancer.gov/).

***

### [*DriverDBv4*](http://140.128.69.158/)

Database for human cancer driver gene research
Link to [Help Page](http://driverdb.bioinfomics.org/help/).

***

### [*List of PPI resources*](https://startbioinfo.org/cgi-bin/simpleresources.pl?tn=PPI_AR)

A list of 375 protein-protein interaction resources were compiled through extensive literature search. Basic features, URL, publication date, and corresponding article(s) for each resource are also listed by browsing through their web-pages.

***

### [*Cancer cell maps*](https://ccmi.org/cell-maps)

![Cancer cell maps!](/assets/img/cancer-cell-maps.png)

Selected set of human cancer focused pathways.

***

### [*Tumour-suppressor Gene Database (TSGene)*](https://bioinfo.uth.edu/TSGene/)

![Tumour-suppressor Gene Database (TSGene)!](/assets/img/tsgene.png)

Comprehensive resource for pan-analysis of human tumour suppressor genes (TSGs).
Link to [Help Page](https://bioinfo.uth.edu/TSGene/tutorial.cgi?csrt=1241548866957659715).

***

### [*COSMIC*](https://cancer.sanger.ac.uk/cosmic)

![COSMIC!](/assets/img/cosmic.png)

Catalogue of somatic mutations in cancer.
Link to [Help Page](https://cancer.sanger.ac.uk/cosmic/help).

***

### [*Cancer Hotspots*](https://www.cancerhotspots.org/#/home)

![Cancer Hotspots!](/assets/img/cancer-hotspots.png)

Resource for statistically significant recurrent mutations in cancer.

***

### [*CancerMine*](http://bionlp.bcgsc.ca/cancermine/)

Literature-mined database of drivers, oncogenes and tumour suppressors in cancer.

***

### [*Cancer cell metabolism gene DB*](https://bioinfo.uth.edu/ccmGDB/)

Comprehensive annotation resource for cell metabolism genes in cancer.
Link to [Help Page](https://bioinfo.uth.edu/ccmGDB/help.html?csrt=1241548866957659715).

***

### [*DisGeNET*](https://www.disgenet.org/)

![DisGeNET!](/assets/img/disgenet.png)

Discovery platform containing one of the largest publicly available collections of genes and variants associated to human diseases
Link to [Help Page](https://www.disgenet.org/help).

***

### [*Cancer 3D*](http://cancer3d.org/search)

Patterns of mutations in cancer. Database provides an open and user-friendly way to analyse cancer missense mutations in the context of structures of proteins they are found in and in relation to patients gener and age.
Link to [Tutorial](http://cancer3d.org/tutorial).

***

### [*GREIN*](http://www.ilincs.org/apps/grein/)

![GREIN!](/assets/img/grein.png)

GEO RNAseq Experiments Interactive Navigator. Interactive web platform that provides user-friendly options to explore and analyse GEO RNAseq data.

***

### [*DISCO - deeply integrated human single-cell omics data*](https://www.immunesinglecell.org/)

![DISCO!](/assets/img/disco.png)

Massive database. Three tools available on there too for scRNAseq data. CELLiD: cell type annotation, CellMapper:data transfer, and scEnrichment: GSEA.
Link to [Vignette/Tutorial](https://immunesinglecell.org/vignette/docs/DISCO/overview).

***

### [*cellTypist*](https://buff.ly/44YEu47) 

![cellTypist!](/assets/img/cell-typist.png)

It is a cell annotation tool but also have a database for the markers.
Link to [Tutorials](https://www.celltypist.org/tutorials).

***

### [*Cellxgene*](https://cellxgene.cziscience.com/)

![Cellxgene!](/assets/img/cellxgene.png)

Download and/or visually explore reference-quality data to understand the functionality of human tissues at the cellular level with Chan Zuckerberg CELL by GENE Discover (CZ CELLxGENE Discover).
Users can search for data collections, browse data sets, explore single cell gene expression, and obtain information about specific cell types. Really neat tool!
[Help & Documentations Page](https://cellxgene.cziscience.com/docs/01__CellxGene).

***

### [*IMMUcan Database*](https://immucanscdb.vital-it.ch/)

![IMMUcan Database!](/assets/img/immucan.png)

A fully integrated scRNA-seq database exclusively dedicated to human cancer and accessible to nonspecialists. IMMUcan scDB encompasses 144 datasets on 56 different cancer types, annotated in 50 fields containing precise clinical, technological, and biological information.

***

### [*scPortalen Database*](https://single-cell.riken.jp/SCPortalen_Database/Home/)

![scPortalen Database!](/assets/img/scportalen.png)

This database features integration of single cell metadata, cell images and sequence information. Rhe platform is divided into two parts, the single cell dataset generated by the research team, and the single cell dataset published in peer-reviewed papers.

***

### [*Cancer Single-cell Expression Map*](https://ngdc.cncb.ac.cn/cancerscem/)

![Cancer Single-cell Expression Map!](/assets/img/cancer-single-cell-expression-map.png)

Public database dedicating to collecting, analysing, and visualising single-cell RNAseq data of human cancers.
Link to [Help Page](https://ngdc.cncb.ac.cn/cancerscem/help).

***

### [*TISCH2: Tumour Immmune Single-cell Hub 2*](http://tisch.comp-genomics.org/home/)

![TISCH2!](/assets/img/tisch2.png)

Tumor Immune Single-cell Hub 2 (TISCH2) is a scRNA-seq database focusing on tumor microenvironment (TME). TISCH2 provides detailed cell-type annotation at the single-cell level, enabling the exploration of TME across different cancer types.
Link to [documentation Page](http://tisch.comp-genomics.org/documentation/).

***

### [*Kaplan-Meier Plotter*](https://kmplot.com/analysis/index.php?p=background)

![Kaplan-Meier Plotter!](/assets/img/km-plotter.png)

The Kaplan Meier plotter is capable of assessing the correlation between the expression of all genes (mRNA, miRNA, protein, & DNA) and survival in 35k+ samples from 21 tumor types. Applied statistical tools include Cox proportional hazards regression and the computation of the False Discovery Rate. With 20,000 analyses per day, the KM-plotter is a worldwide reference for the discovery and validation of survival biomarkers.

***

### [*ROC Plotter*](https://rocplot.com)

![ROC Plotter!](/assets/img/roc-plotter.png)

The ROC plotter is capable of linking gene expression and therapy response using transcriptome-level data of breast, ovarian, and colorectal cancer patients and glioblastomas. The custom plotter generates a ROC plot for user-uploaded data.

***

### [*muTarget*](https://mutarget.com)

![muTarget!](/assets/img/mutarget.png)

muTarget is a cancer biomarker / target discovery tool with two major functions

1) With a "Genotype" run one can identify gene(s) showing altered expression in samples harbouring a mutated input gene. This option is useful in case one searches new drug targets in a cohort of patients with a given mutation.

2) With a "Target" run one can identify mutations resulting in expression change in the input gene. This option is useful in case one has a drug target gene, and patient cohorts with enriched expression is the question.

***

### [*Cancer Hallmarks*](https://cancerhallmarks.com)

![Cancer Hallmarks!](/assets/img/cancer-hallmarks.png)

What are the Hallmarks of Cancer?

The "Hallmarks of Cancer" concept provides a framework for understanding fundamental organizing principles common to various cancers. Understanding hallmarks can benefit cancer prevention, diagnosis, and treatment development by summarizing functional and metabolic commonalities underlying malignant transformation and progression.

What can CancerHallmarks do?

We established a consensus list of cancer hallmark genes by merging 6,763 genes from available mapping resources. CancerHallmarks.com enhances the utility of the hallmark concept as an effective organizational tool by funneling genes to biological functions.

***

### [*TNMplot: differential gene expression analysis in Tumour, Normal, and Metastatic tissues*](https://tnmplot.com)

![TNMplot!](/assets/img/tnm-plot.png)

Realy nice tool! Performs a whole range of pan-cancer analyses and provides multiple different outputs such as dot plots, bar charts, correlation plots, box plots and more!

***

***

# Cancer Atlases

Here you will find links to various cancer atlases.

***

***

### [*St Judes*](https://www.stjude.cloud/)

![St Judes!](/assets/img/st-judes.png)

St Judes has a collection of tools:
Link to the [Help Guides](https://university.stjude.cloud/docs/).

[St Judes Genomics platform](https://platform.stjude.cloud/)
Has data browser - one of the wworlds most comprehensive repositories of pediatric cancer genomics data. Has analysis workflows to analyse the genomics data.

![St Judes Genomics platform!](/assets/img/genomics-platform.png)

[PECAN](https://pecan.stjude.cloud/) - curated pediatric cancer genomics data including variants, mutational signatures, and gene expression data in addition to histological slide images from ~9000 hematological, CAN, and non-CNS solid tumour patient samples.

![PECAN!](/assets/img/pecan.png)

[Xenograft model systems tool](https://models.stjude.cloud/) - Explore patient derived xenograft data and cell lines generated at st Jude to enhance basic research and speed translation to the clinic.

![Xenograft model systems tool!](/assets/img/model-systems.png)

Theres also a visualisation community that allows the creation and sharing of figures!

***

### [*Human Cell Atlas (HCA)*](https://www.humancellatlas.org/) 

![Human Cell Atlas (HCA)!](/assets/img/human-cell-atlas.png)

A database of cellular reference maps with position, function and characteristics of every cell type in the human body.
[Heres the link to the data site](https://data.humancellatlas.org/ data site).
[Link to the HCA guide](https://data.humancellatlas.org/guides).

***

### [*Cambridge Cell Atlas*](https://www.cambridgecellatlas.org/)

![Cambridge Cell Atlas!](/assets/img/cambridge-cell-atlas.png)

Portal of the human cell atlas. Analyses, visualises and provides tools for exploration of singlke cell RNAseq data generated by HCA. Nice tool.

***

### [*Single Cell Expression Atlas*](https://www.ebi.ac.uk/gxa/sc/home)

![Single Cell Expression Atlas!](/assets/img/single-cell-expression-atlas.png)

Run by the EBI. Can search genes, cell types, organs and diseases across 21 species, 355 studies and 10,505,726 cells. 
Link to the [Help Page](https://www.ebi.ac.uk/gxa/sc/help.html).

***

### [*Curated Cancer Cell Atlas (3CA)*](https://www.weizmann.ac.il/sites/3CA/)

![Curated Cancer Cell Atlas (3CA)!](/assets/img/3ca.png)

Collected, annotated, and analysed cancer scRNAseq datasets.

***

***

# Cancer Specific Tools

***

***

### [*Leukaemia Gene Database (LeGenD)*](https://www.bioinformatics.org/legend/legend.htm)

![Leukaemia Gene Database (LeGenD!](/assets/img/legend.png)

Database of leukaemia genes (LeGenD) developed to help biological and medical sciences community to easily access all information on the genes that are involved in leukaemia.
Link to [Help Page](http://www.bioinformatics.org/legend/help.htm).

***

### [*Pancreatic Cancer Gene Database (PCGDB)*](https://www.bioinformatics.org/pcgdb/)

![Pancreatic Cancer Gene Database (PCGDB)!](/assets/img/pcgdb.png)

Provides information on the genes that are involved in pancreatic cancer and this data is targeted to help the biological and medical sciences community for easier access of the latest information on genes causing pancreatic cancer.
Link to [Help Page](https://www.bioinformatics.org/pcgdb/fact%20page.htm).

***

### [*Mesothelioma Cancer Advocacy*](https://mesothelioma.net/mesothelioma/)

![Mesothelioma.net!](/assets/img/mesothelioma.png)

Mesothelioma.net is an advocacy and support group dedicated to providing all the latest in cancer research, treatment, and aid.  Our team has worked diligently with health professionals to compile fact-checked and physician approved information regarding this disease and how it can be treated.

***
