---
type: page
title: General Bioinformatics Resources
author: Alex Gibbs
---

***THIS PAGE IS UNDER CONSTRUCTION***

This page is intended to provide researchers with useful tools to aid their research.
These tools may be a webpage link to certain tools (such as a website that allows you to search for information related to your favourite gene) or softwares.

***

### [*RegNetwork: Regulatory Network Repository*](https://regnetworkweb.org/home.jsp)

![RegNetwork!](/assets/img/regnetwork.png)

RegNetwork is a data repository of five-type transcriptional and posttranscriptional regulatory relationships for human and mouse.
Link to the [About page](https://regnetworkweb.org/about.jsp). 

***

### [*HATA: The Human Adipose Tissue Atlas*](https://vimrg.app.vumc.org/vimrg/)

![HATA!](/assets/img/HATA.png)

The Koethe research group have generated a single cell RNAseq and CITE-seq dataset consisting 59 HIV+ (20 non-diabetic, 19 pre-diabetic, 20 diabetic) and 32 HIV- diabetic patients.

Very nice tool.

***

### [*ENCODE*](https://www.encodeproject.org/)

![ENCODE!](/assets/img/encode.png)

The ENCODE Portal, developed and maintained by the Data Coordination Center (ENCODE DCC), is the canonical source for all experimental metadata and data from ENCODE and associated projects. The ENCODE Portal contains raw and ground-level analysis data generated by participating mapping centers using a wide-range of assays (integrative analysis data are available at the SCREEN portal. The Portal also stores records of the materials and methods used to perform the assays and subsequent analysis. No account is needed to view or download released data.

Link to the [Tutorial page](https://www.encodeproject.org/help/getting-started/). 

***


### [*IHEC: The International Human Epigenome Consortium*](https://epigenomesportal.ca/ihec/index.html)

![IHEC!](/assets/img/ihec.png)

The International Human Epigenome Consortium (IHEC) makes available comprehensive sets of reference epigenomes relevant to health and disease. The IHEC Data Portal can be used to view, search and download the data already released by the different IHEC-associated projects.

Link to the [Help page](https://epigenomesportal.ca/ihec/help.html). 

***

### [*Blueprint Project*](http://blueprint-data.bsc.es/#!/)

![blueprint!](/assets/img/blueprint.png)

Explore the behaviour of your candidate genes, pathways or genome regions across the Blueprint datasets.

The Blueprint project is providing in this release epigenomic analysis from 1039 samples to the scientific community. Their associated epigenomes are characterized by: gene and transcript expression (from RNA-Seq experiments), hyper and hypo methylated regions (derived from WGBS experiments), chromatin accessibility (DNAseI-Seq), and 9 Histone marks binding activity (ChIP-Seq)

Use this tool to explore the behaviour of your genes and pathways in the BLUEPRINT analysis for 62 cellular types and cellular lines.

Link to the [knowing the interface page](http://blueprint-data.bsc.es/#!/first-steps).
Link to the [knowing the charts page](http://blueprint-data.bsc.es/#!/charts-help).

***

### [*EpiFactors*](https://epifactors.autosome.org/)

EpiFactors is a database for epigenetic factors, corresponding genes and products.

Link to the [help and information page](https://epifactors.autosome.org/description).

***

### [*CistromeDB*](http://cistrome.org/db/#/)

![Cistrome!](/assets/img/cistrome.png)

[CistromeDB Toolkit](http://dbtoolkit.cistrome.org/) - Check what factors regulate your gene of interest, what factors bind in your interval or have a significant binding overlap with your peak set.

[Cistrome-GO](http://go.cistrome.org/) - Predict the function of transcription factor perturbed expression from Transcription Factor ChIP-seq data.

Link to the [about page](http://cistrome.org/db/#/about).
Link to the [documentation page](http://cistrome.org/db/#/documents).

***

### [*ReMap2022*](https://remap2022.univ-amu.fr/)

ReMap is a large scale integrative analysis of DNA-binding experiments for Homo sapiens, Mus musculus, Drosophila melanogaster and Arabidopsis thaliana transcriptional regulators. the catalogues are the results of the manual curation of ChIP-seq, ChIP-exo, DAP-seq from public sources (GEO, ENCODE, ENA).

A tour of the site can be found at the top right-hand side of the web page.

***

### [*ChIP-Atlas*](https://chip-atlas.org/)

![ChIP-Atlas!](/assets/img/chip-atlas.png)

A data-mining suite for exploring epigenomic landscapes by fully integrating 419,000 ChIP-seq, ATAC-seq and Bisulfate-seq experiments.

[Cistrome-GO](http://go.cistrome.org/) - Predict the function of transcription factor perturbed expression from Transcription Factor ChIP-seq data.

Link to the [documentation page](https://github.com/inutano/chip-atlas/wiki).

***

### [*FANTOM5: Functional Annotation of the Mammalian Genome*](https://fantom.gsc.riken.jp/5/)

![FANTOM5!](/assets/img/fantom5.png)

The FANTOM5 project is a worldwide collaborative project aiming at identifying all functional elements in mammalian genomes.

Link to the [FAQ page](https://fantom.gsc.riken.jp/faq/).

***


### [*FORGEdb*](https://forgedb.cancer.gov/)

![FORGEdb!](/assets/img/forgedb.png)

FORGEdb is a tool for studying regulatory genomic variants, the "dark matter" of the genome. FORGEdb annotates these little-known variants with eQTLs, chromatin looping, epigenetic  marks, TF binding sites, and more!

Link to the [about page](https://forgedb.cancer.gov/about/).

***

### [*CuratedAtlasQueryR*](https://stemangiola.github.io/CuratedAtlasQueryR/index.html)

![CuratedAtlasQueryR!](/assets/img/curatedatlasqueryr.png)

CuratedAtlasQueryR is a query interface in R that allows the programmatic exploration and retrieval of the harmonised, curated and reannotated CELLxGENE single-cell human cell atlas. Data can be retrieved at cell, sample, or dataset levels based on filtering criteria.

***

### [*ezSinglecell*](https://immunesinglecell.org/ezsc/)

ezSinglecell is an interactive and easy-to-use application for analysing various single-cell and spatial omics data trypes without requiring prior programming knowledge. It combines the best-performing public methods for in-depth data analysis and interactive data visualisation. It consists of five modules, each designed to be a comprehensive workflow for one data type or task. It also allows crosstalk between different modules within a defined interface.

Link to the [help page](https://immunesinglecell.org/ezsc/) which contains manuals for each module.

***

### [*Cellar*](https://cellar.cmu.hubmapconsortium.org/app/cellar)

![Cellar!](/assets/img/cellar.png)

Cellar is an interactive tool for analysing single-cell omics data. The tool supports, but is not limited to, scRNA-seq, scATAC-seq, CODEX, SNARE-seq, sciRNA-seq, and visium. It supports pre-processing, dimensionality reduction, clustering, DE gene testing, enrichment analysis, culster and gene visualisation modules, projection to spatial tiles, label transfer, and semi-supervised clustering among others.

Link to the [documentation page](https://euxhenh.com/cellar/).
There are also [video tutorials](https://www.youtube.com/playlist?list=PL5sLSLkTYpWgfBQ0M8ObfBIqDMAzx0-D2) available.

***


### [*ASAP: Automated Single-cell Analysis Portal*](https://asap.epfl.ch/)

![ASAP!](/assets/img/asap.png)

ASAP is a web-based collaborative portal that allows the analysis of single-cell transcritpomic data.

Link to the [tutorial page](https://asap.epfl.ch/home/tutorial).

***

### [*iSEE*](https://bioconductor.org/packages/release/bioc/vignettes/iSEE/inst/doc/basic.html)

![iSEE!](/assets/img/isee.png)

iSEE is a bioconductor package that provides an interactive Shiny-based graphical user interface for exploring data stored in *SummarizedExperiment* objects.

Link to the [installation page](https://bioconductor.org/packages/3.19/bioc/html/iSEE.html) that contains HTML pages and R Scripts.

***

### [*VISION*](https://github.com/YosefLab/VISION)

![VISION!](/assets/img/vision.png)

VISION is an R package that aids in the interpretation of single-cell RNA-seq data by selecting for gene signatures which describe coordinated variation between cells. Whilst the software only requires an expression matric and a signature library (available in online databases), it is also desitnged to integrate into existing scRNA-seq analysis pipelines by taking advantage of precomputed dimensionality reductions, trajectory inferences or clustering results. The results of this analysios are makde available through a dynamic web-app which can be shared with collaborators without requiring them to install any additional softwares.

Link to the [getting started vignette](https://yoseflab.github.io/VISION/articles/VISION-vignette.html).

***

### [*SComatic*](https://github.com/cortes-ciriano-lab/SComatic)

SComatic is a tool that provides functionalities to detect somatic single-nucleotide mutations in high-throughput single-cell genomics and transcriptomics data sets, such as single-cell RNA-seq and single-cell ATAC-seq.

***

### [*scRNA-tools*](https://www.scrna-tools.org/)

![scRNA-tools!](/assets/img/scrna-tools.png)

The scRNA-tools database is a catalogue of tools for analysing single-cell RNA sequencing data. Really nice website that can be your one-stop shop to find the right tool for the analysis you are looking to perform!

Link to the [FAQ page](https://www.scrna-tools.org/faqs).

***

### [*Kaplan-Meier Plotter*](https://kmplot.com)

![Kaplan-Meier Plotter!](/assets/img/km-plotter.png)

The Kaplan Meier plotter is capable of assessing the correlation between the expression of all genes (mRNA, miRNA, protein, & DNA) and survival in 35k+ samples from 21 tumor types. Applied statistical tools include Cox proportional hazards regression and the computation of the False Discovery Rate. With 20,000 analyses per day, the KM-plotter is a worldwide reference for the discovery and validation of survival biomarkers.

***

### [*InnateDB*](https://www.innatedb.com/index.jsp)

![InnateDB!](/assets/img/innatedb.png)

InnateDB is publicly available database of the genes, proteins, experimentally-verified interactions and signaling pathways involved in the innate immune response of humans, mice and bovines to microbial infection. The database captures an improved coverage of the innate immunity interactome by integrating known interactions and pathways from major public databases together with manually-curated data into a centralised resource. To date, 18,780 interactions have been manually curated by the InnateDB curation team. The database can be mined as a knowledgebase or used with our integrated bioinformatics and visualization tools for the systems level analysis of the innate immune response. 

A website tutorial can be found at the top right of the home page.

***

### [*ROC Plotter*](https://rocplot.com)

![ROC Plotter!](/assets/img/roc-plotter.png)

The ROC plotter is capable of linking gene expression and therapy response using transcriptome-level data of breast, ovarian, and colorectal cancer patients and glioblastomas. The custom plotter generates a ROC plot for user-uploaded data.

***

### [*muTarget*](https://mutarget.com)

![muTarget!](/assets/img/mutarget.png)

muTarget is a cancer biomarker / target discovery tool with two major functions

1) With a "Genotype" run one can identify gene(s) showing altered expression in samples harbouring a mutated input gene. This option is useful in case one searches new drug targets in a cohort of patients with a given mutation.

2) With a "Target" run one can identify mutations resulting in expression change in the input gene. This option is useful in case one has a drug target gene, and patient cohorts with enriched expression is the question.

***

### [*TNMplot: differential gene expression analysis in Tumour, Normal, and Metastatic tissues*](https://tnmplot.com)

![TNMplot!](/assets/img/tnm-plot.png)

Realy nice tool! Performs a whole range of pan-cancer analyses and provides multiple different outputs such as dot plots, bar charts, correlation plots, box plots and more!

***
